src/resolver.cpp

```diff
@@ -235,211 +222,95 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
-           << "result: " << result.GetPathString();
-    logger_->debug(logMsg.str());
-  }
+/*
+ * Write
+ *
+ * We don't currently support writes to OpenAssetIO entity references.
+ * In order to call register when the ArAsset is closed, we'll need
+ * not resolve in _ResolveForNewAsset, and pass the entity reference
+ * through.
+ */
 
-  return result;
+std::string UsdOpenAssetIOResolver::_CreateIdentifierForNewAsset(
+    const std::string &assetPath, const ArResolvedPath &anchorAssetPath) const {
+  if (manager_->isEntityReferenceString(assetPath)) {
+    std::string message = "Writes to OpenAssetIO entity references are not currently supported ";
+    message += assetPath;
+    logger_->critical(message);
+    return "";
+  }
+  return ArDefaultResolver::_CreateIdentifierForNewAsset(assetPath, anchorAssetPath);
 }
 
-// TODO(DF): Implement for publishing workflow.
 ArResolvedPath UsdOpenAssetIOResolver::_ResolveForNewAsset(const std::string &assetPath) const {
-  auto result = ArDefaultResolver::_ResolveForNewAsset(assetPath);
-
-  logger_->debug(TF_FUNC_NAME());
-  logger_->debug("  assetPath: " + assetPath);
-  logger_->debug("  result: " + result.GetPathString());
-
-  return result;
+  if (manager_->isEntityReferenceString(assetPath)) {
+    std::string message = "Writes to OpenAssetIO entity references are not currently supported ";
+    message += assetPath;
+    logger_->critical(message);
+    return ArResolvedPath("");
+  }
+  return ArDefaultResolver::_ResolveForNewAsset(assetPath);
 }
 
-/* Asset Operations*/
+/*
+ * Pass-through asset operations
+ *
+ * These methods are simply relayed to the ArDefaultResolver. There may
+ * be interest in fetching data for some of these from the manager, but
+ * we don't have a real use case just yet. Doing so increases complexity
+ * as we'd need to return both the resolved path _and_ the original
+ * entity reference from _Resolve, so we could make queries in these
+ * methods. We'll need this for publishing operations, but avoiding that
+ * overhead for the more common (and hot) read case is critical.
+ *
+ */
 std::string UsdOpenAssetIOResolver::_GetExtension(const std::string &assetPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  assetPath: " << assetPath;
-    logger_->debug(logMsg.str());
-  }
-
-  auto result = catchAndLogExceptions(
-      [&] {
-        // TODO(DF): Give the manager a chance to provide the file extension.
-        return ArDefaultResolver::_GetExtension(
-            locationInManagerContextForEntity(manager_, readContext_, assetPath));
-      },
-      logger_, fnName);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result;
-    logger_->debug(logMsg.str());
-  }
-
-  return result;
+  return ArDefaultResolver::_GetExtension(assetPath);
```

Comment:
[question] Is this is an example where it's reasonable that "the old approach is only done in `_ResolveForNewAsset`", since it doesn't make much sense to give the manager a chance to override the file extension reported for a pre-existing file?

Reply:
Yeah, given we only support loading from files at this point, and the file is there...